app/models/signature.rb

```diff
@@ -266,6 +265,29 @@ def rate(window = 5.minutes)
     petition.signatures.where(ip_address: ip_address, created_at: period).count
   end
 
+  def email_count
+    super || 0
+  end
+
+  def email_threshold_reached?
+    email_count >= 5
+  end
+
+  def duplicate?
+    matcher = Signature.where(email: email, petition_id: petition_id)
+    matcher = matcher.where("signatures.id != ?", id) unless new_record?
+    existing_email_address_count = matcher.count
+    return false if existing_email_address_count == 0
+    return true if existing_email_address_count > 1
+    existing_signature = matcher.first
+    return true if (existing_signature.name.strip.downcase == name.strip.downcase)
+    if (existing_signature.postcode.gsub(/\s+/,'').downcase !=
+        postcode.gsub(/\s+/,'').downcase)
+      return true
+    end
+    false
+  end
```

Comment:
Total nitpick but I think the last few lines here (after `existing_signature = matcher.first`) could be made clearer - we've mixed the styles guard clause vs. full if statement and I think it would be clearer if it was all one style so either:

```ruby
return true if (existing_signature.name.strip.downcase == name.strip.downcase)
return true if (existing_signature.postcode.gsub(/\s+/,'').downcase != postcode.gsub(/\s+/,'').downcase)
return false
```

or:

```ruby
if (existing_signature.name.strip.downcase == name.strip.downcase ||
   (existing_signature.postcode.gsub(/\s+/,'').downcase != postcode.gsub(/\s+/,'').downcase)
  true
else
  false
end
```

Or some 3rd option that you feel works.